### DISCUSSION TITLE
encountered error in rb-mpi (1.2.5-preview)

### DISCUSSION THREAD

**Discussion by FamLiuLab** [revbayes/revbayes#626] [Q&A]
I compiled a rb-mpi on my mac using source code, which resulted the newest executable version "1.2.5-preview". However, in the first test run using a previously working script in 1.2.1, the following error occurs: "Error:	Setup includes a move for sampled ancestors but the corresponding tree distribution doesn't allow sampled ancestors". The corresponding command line is "moves.append( mvCollapseExpandFossilBranch(fbd_tree, origin_time, weight=6.0) )". I subsequently re-run the same script in 1.2.1, it has no errors at all.  I would retreat to 1.2.1 if I could. The problem is that the only 1.2.1 I currently have is a non-mpi version which cannot handle my large dataset and complete analysis within a reasonable time (6000 hours!). Thank you in advance, and any insight will be much appreciated!

**Comment by davidcerny** [comment]
Hi @FamLiuLab, this has nothing to do with the fact you're using the MPI version of RevBayes – it literally just means you're trying to sample ancestors, but the distribution from which you're drawing your time tree doesn't allow those. My guess is you're drawing `fbd_tree` from `dnBDSTP` with `r=1`? If so, you need to either (1) make sure to set `r=0`, or (2) just use `dnFBDP` instead.

For some context, see https://github.com/revbayes/revbayes/issues/423, https://github.com/revbayes/revbayes/pull/440. The `r` argument stands for the probability of removing a lineage from the process upon sampling it; if you set it to 1, you're never going to get sampled ancestors, which is what the error message is trying to communicate. While your script may have seemingly "worked" in v1.2.1, if you carefully examine its output, you'll find out no ancestors were ever sampled!
